src/steamship/agents/service/agent_service.py

```diff
@@ -165,6 +202,9 @@ def run_agent(self, agent: Agent, context: AgentContext):
             logging.info(f"Emitting via function: {func.__name__}")
             func(action.output, context.metadata)
 
+        # Finally, mark this block as responded to.
+        last_user_message.set_response_sent(True)
```

Comment:
i wonder what the implications are for transports?  If the agent think that a message has been successfully sent, but the transport fails to deliver, will we have issues?  Does the `emit_func` need a different signature to allow for that possibility?

Reply:
Since our PackageService failure model is `throw on fail`, then a failed transport  would result in the message not getting stamped as having been delivered.

I think Step 1 for defining something more nuanced would probably be to define a different set of semantics around `emit_fn`, yeah.